qemu/tests/virtio_fs_group_permission_access.py

```diff
@@ -0,0 +1,148 @@
+
+from provider import virtio_fs_utils, win_driver_utils
+
+
+@error_context.context_aware
+def run(test, params, env):
+    """
+    Start virtiofsd with the user who has permission to access the directory
+    of another user, then test whether the shared directory works well or not.
+
+    1) Create two users( e.g. u1 and u2) on host.
+    2) Change permission of u1 home directory, give all write permission.
+    3) Change user u2's supplementary group to u1.
+    4) Start virtiofsd under user u2
+       with the option "--shared-dir /home/u1/virtio_fs_test".
+    5) Boot a guest on the host.
+    6) If guest is windows, start the viofs service and reboot.
+    7) If the guest is linux, mount the file system.
+    8) Run the basic io test and folder accessing test.
+    9) After test, clear the environment.
+
+    :param test: QEMU test object
+    :param params: Dictionary with the test parameters
+    :param env: Dictionary with test environment
+    """
+    add_user_cmd = params.get("add_user_cmd")
+    del_user_cmd = params.get("del_user_cmd")
+    username = params.objects("new_user")
+    windows = params.get("os_type", "windows") == "windows"
+    driver_name = params.get("driver_name")
+    fs_dest = params.get("fs_dest")
+    fs_target = params.get("fs_target")
+    cmd_run_virtiofsd = params.get("cmd_run_virtiofsd")
+
+    # nfs config
+    setup_local_nfs = params.get('setup_local_nfs')
+
+    fs_source = params.get('fs_source_dir')
+    host_session = None
+    guest_session = None
+    vm = None
+    nfs_local = None
+
+    try:
+        for _username in username:
+            cmd = add_user_cmd % _username
+            process.run(cmd)
+        user_one, user_two = username[0], username[-1]
+        # create the folder before daemon running
+        shared_dir = os.path.join("/home/" + user_one, fs_source)
+        if not os.path.exists(shared_dir):
+            process.system("runuser -l " + user_one + " -c 'mkdir -p " +
+                           shared_dir + "'")
+
+        if setup_local_nfs:
+            # delete the slash at the end
+            params["nfs_mount_dir"] = shared_dir[:-1]
+            nfs_local = nfs.Nfs(params)
+            nfs_local.setup()
+
+        # change permission of u1 home directory
+        output = process.system_output("chmod -R 777 /home/%s" % user_one)
+        error_context.context(output, test.log.info)
+        # change user u2's supplementary group to u1
+        output = process.system_output("usermod -G %s %s" % (user_one,
+                                                             user_two))
+        error_context.context(output, test.log.info)
+
+        # set fs daemon config
+        sock_path = os.path.join("/home/" + user_two,
+                                 '-'.join(('avocado-vt-vm1', 'viofs',
+                                           'virtiofsd.sock')))
+        # create the file
+        with open(sock_path, "w") as fd:
```

Comment:
For windows guest, qemu-kvm process quit as ' [qemu output] qemu-kvm: -chardev socket,id=char_virtiofs_fs: Unable to parse FD number
'